<?xml version="1.0" encoding="UTF-8"?>

<launch>
	<arg name="world"  default="small_office_inventory"/>
	<arg name="multi" default=""/>
	<arg name="gzpose" default="-x 0.0 -y 0.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/>
	<arg name="gui" default="true"/>
	<arg name="debug" default="false"/>
	<arg name="recording" default="false"/>
	<arg name="extra_gazebo_args" default=""/>
	<arg name="use_moveit_camera" default="false"/>
	<arg name="use_dynamic_footprint" default="false"/>
	<arg name="global_planner" default="global_planner"/>

	<node pkg="path_finding" type="turn_towards.py" name="turn_towards_server"/>

	<node pkg="path_finding" type="pickup_objects.py" name="pickup_objects" output="screen" required="true"/>
	<rosparam file="$(find my_simulation)/params/$(arg world).yaml" />
	<rosparam param="objects_to_pickup">["a", "b", "c"]</rosparam>

	<include file="$(find my_simulation)/launch/my_simulation.launch" >
		<arg name="world"  value="$(arg world)" />
		<arg name="global_planner" value="$(arg global_planner)"/>
		<arg name="multi" value="$(arg multi)" />
		<arg name="gzpose" value="$(arg gzpose)" />
		<arg name="gui" value="$(arg gui)" />
		<arg name="debug" value="$(arg debug)" />
		<arg name="recording" value="$(arg recording)" />
		<arg name="extra_gazebo_args" value="$(arg extra_gazebo_args)" />
		<arg name="use_moveit_camera" value="$(arg use_moveit_camera)" />
		<arg name="use_dynamic_footprint" value="$(arg use_dynamic_footprint)" />
	</include>

	<node pkg="items_locating" type="find_object.py" name="find_object"/>
	<node pkg="path_finding" type="plan_path.py" name="plan_path" output="screen"/>


</launch>

